src/include/rbd/librbd.hpp

```diff
@@ -302,10 +309,10 @@ class CEPH_RBD_API Image
   int flatten();
   int flatten_with_progress(ProgressContext &prog_ctx);
   /**
-   * Returns a pair of poolname, imagename for each clone
-   * of this image at the currently set snapshot.
+   * Returns a structure of poolname, imagename, imageid and trash flag
+   * for each clone of this image at the currently set snapshot.
    */
-  int list_children(std::set<std::pair<std::string, std::string> > *children);
+  int list_children(std::vector<librbd::child_info_t> *children);
```

Comment:
You cannot change the public API -- you can only change the private API that is implemented in "src/librbd/internal.cc"

Reply:
Oh, sorry, at first I thought `int list_children(std::set<std::pair<std::string, std::string> > *children)` was unused, so i changed it directly. Actually there may be other external APPs used it. Already modified, thanks.